packages/docmanager-extension/src/index.tsx

```diff
@@ -93,12 +94,77 @@ namespace CommandIDs {
  */
 const docManagerPluginId = '@jupyterlab/docmanager-extension:plugin';
 
+/**
+ * A plugin to open documents in the main area.
+ *
+ */
+const openerPlugin: JupyterFrontEndPlugin<IDocumentWidgetOpener> = {
+  id: '@jupyterlab/docmanager-extension:opener',
+  autoStart: true,
+  provides: IDocumentWidgetOpener,
+  activate: (app: JupyterFrontEnd) => {
+    const { shell } = app;
+    return {
+      open(widget: IDocumentWidget, options?: DocumentRegistry.IOpenOptions) {
+        if (!widget.id) {
+          widget.id = `document-manager-${++Private.id}`;
+        }
+        widget.title.dataset = {
+          type: 'document-title',
+          ...widget.title.dataset
+        };
+        if (!widget.isAttached) {
+          shell.add(widget, 'main', options || {});
+        }
+        shell.activateById(widget.id);
+        this.opened.emit(widget);
+      },
+      get opened() {
+        if (this._signal) {
+          return this._signal;
+        }
+        this._signal = new Signal<IDocumentWidgetOpener, IDocumentWidget>(this);
+        return this._signal;
+      }
+    };
```

Comment:
Let's freeze the object to forbid tempering:

```suggestion
    return Object.freeze({
      open(widget: IDocumentWidget, options?: DocumentRegistry.IOpenOptions) {
        if (!widget.id) {
          widget.id = `document-manager-${++Private.id}`;
        }
        widget.title.dataset = {
          type: 'document-title',
          ...widget.title.dataset
        };
        if (!widget.isAttached) {
          shell.add(widget, 'main', options || {});
        }
        shell.activateById(widget.id);
        this.opened.emit(widget);
      },
      get opened() {
        if (this._signal) {
          return this._signal;
        }
        this._signal = new Signal<IDocumentWidgetOpener, IDocumentWidget>(this);
        return this._signal;
      }
    });
```